src/components/messages/Message/MessageItem.tsx

```diff
@@ -9,8 +9,19 @@ interface IMessageItemProps {
   message: ChatClientTypes.Message
   index: number
   peer?: string
+  previousMessageAccount?: string
+  nextMessageAccount?: string
 }
-export const MessageItem: React.FC<IMessageItemProps> = ({ messages, message, index, peer }) => {
+
+type ILocalMessageType = 'first' | 'last' | 'same' | 'solo'
+export const MessageItem: React.FC<IMessageItemProps> = ({
+  messages,
+  message,
+  index,
+  peer,
+  previousMessageAccount,
+  nextMessageAccount
+}) => {
```

Comment:
The responsibility of knowing the next and previous message, and changing styling accordingly should be that of a parent container. I'm not saying this is a total blocker, but if we keep this in like this, it will be refactored at some point when new requirements come up.

Reply:
Sounds good, I'll take a look at this then. 
The reason why I added this was to maintain the classnames for the children. 
Technically we could group them in another div based on the parents and style them accordingly, but that was leading to a lot of regrouping happening amongst divs since every new message would change the structure of divs and it didnt feel optimal at the time.

Reply:
Noting this down here for later:

`Conversation` component, which would encompass `ConversationBeginning` and all `MessageItems`, this container would structure message. For example, when mapping messages to `MessageItem` components, it would check previous and next messages in the map by using the index provided in the map